tests/validation/tests/v3_api/common.py

```diff
@@ -1828,3 +1829,158 @@ def generate_template_global_role(name, new_user_default=False, template=None):
+    validate_ingress(p_client, cluster, [backup_info["workload"]], host, path)
+
+    # Perform Backup
+    backup = cluster.backupEtcd()
+    backup_info["backupname"] = backup['metadata']['name']
+    wait_for_backup_to_active(cluster, backup_info["backupname"])
+
+    # Get all the backup info
+    etcdbackups = cluster.etcdBackups(name=backup_info["backupname"])
+    backup_info["etcdbackupdata"] = etcdbackups['data']
+    backup_info["backup_id"] = backup_info["etcdbackupdata"][0]['id']
+
+    if backup_mode == "s3":
+        backupfileurl = backup_info["etcdbackupdata"][0]['filename']
+        # Check the backup filename exists in S3
+        parseurl = urlparse(backupfileurl)
+        backup_info["backupfilename"] = os.path.basename(parseurl.path)
+        backup_found = AmazonWebServices().s3_backup_check(
+            backup_info["backupfilename"])
+        assert backup_found, "the backup was not found in the S3 bucket"
+    elif backup_mode == 'filesystem':
+        for node in namespace['nodes']:
+            if 'etcd' not in node.roles:
+                continue
+            get_filesystem_snapshots = 'ls /opt/rke/etcd-snapshots'
+            response = node.execute_command(get_filesystem_snapshots)[0]
+            assert backup_info["etcdbackupdata"][0]['filename'] in response,\
+                "The filename doesn't match any of the files locally"
+    return namespace, backup_info
+
+
+def validate_backup_restore(namespace, backup_info):
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    client = get_user_client()
+    cluster = namespace["cluster"]
+    name = random_test_name("default")
+
+    host = namespace["host"]
+    path = "/name.html"
+    con = [{"name": "test1",
+            "image": TEST_IMAGE}]
+
+    # Create workload after backup
+    testworkload = p_client.create_workload(name=name,
+                                            containers=con,
+                                            namespaceId=ns.id)
+
+    validate_workload(p_client, testworkload, "deployment", ns.name)
+
+    # Perform Restore
+    cluster.restoreFromEtcdBackup(etcdBackupId=backup_info["backup_id"])
+    # After restore, validate cluster
+    validate_cluster(client, cluster, intermediate_state="updating",
+                     check_intermediate_state=True,
+                     skipIngresscheck=False)
+
+    # Verify the ingress created before taking the snapshot
+    validate_ingress(p_client, cluster, [backup_info["workload"]], host, path)
+
+    # Verify the workload created after getting a snapshot does not exist
+    # after restore
+    workload_list = p_client.list_workload(uuid=testworkload.uuid).data
+    print(len(workload_list))
+    assert len(workload_list) == 0, "workload shouldn't exist after restore"
+    return namespace, backup_info
+
+
+def validate_backup_delete(backup_mode, namespace, backup_info):
+    client = get_user_client()
+    cluster = namespace["cluster"]
```

Comment:
for the case of "local" we can validate if on a delete of a backup, it is not listed in the API anymore.